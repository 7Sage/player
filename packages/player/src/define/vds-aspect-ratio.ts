import { safelyDefineCustomElement } from '@vidstack/foundation';

import { AspectRatioElement } from '../ui/aspect-ratio/AspectRatioElement';

safelyDefineCustomElement('vds-aspect-ratio', AspectRatioElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-aspect-ratio': AspectRatioElement;
  }
}
