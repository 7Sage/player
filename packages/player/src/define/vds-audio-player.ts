import './vds-media-ui';

import { safelyDefineCustomElement } from '@vidstack/foundation';

import { AudioPlayerElement } from '../players/audio/AudioPlayerElement';

safelyDefineCustomElement('vds-audio-player', AudioPlayerElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-audio-player': AudioPlayerElement;
  }
}
