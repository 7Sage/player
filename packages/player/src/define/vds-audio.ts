import { safelyDefineCustomElement } from '@vidstack/foundation';

import { AudioElement } from '../providers/audio/AudioElement.js';

safelyDefineCustomElement('vds-audio', AudioElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-audio': AudioElement;
  }
}
