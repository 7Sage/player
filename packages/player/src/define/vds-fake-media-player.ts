import './vds-media-ui';

import { safelyDefineCustomElement } from '@vidstack/foundation';

import { FakeMediaPlayerElement } from '../media/test-utils/FakeMediaPlayerElement.js';

safelyDefineCustomElement('vds-fake-media-player', FakeMediaPlayerElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-fake-media-player': FakeMediaPlayerElement;
  }
}
