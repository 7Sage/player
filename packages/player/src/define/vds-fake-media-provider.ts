import { safelyDefineCustomElement } from '@vidstack/foundation';

import { FakeMediaProviderElement } from '../media/test-utils/FakeMediaProviderElement';

safelyDefineCustomElement('vds-fake-media-provider', FakeMediaProviderElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-fake-media-provider': FakeMediaProviderElement;
  }
}
