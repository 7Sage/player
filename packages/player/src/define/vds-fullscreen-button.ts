import { safelyDefineCustomElement } from '@vidstack/foundation';

import { FullscreenButtonElement } from '../ui/fullscreen-button/FullscreenButtonElement.js';

safelyDefineCustomElement('vds-fullscreen-button', FullscreenButtonElement);

declare global {
  interface HTMLElementTagNameMap {
    'vds-fullscreen-button': FullscreenButtonElement;
  }
}
