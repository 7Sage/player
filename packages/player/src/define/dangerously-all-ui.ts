import './vds-aspect-ratio.js';
import './vds-fullscreen-button.js';
import './vds-gesture.js';
import './vds-mute-button.js';
import './vds-play-button.js';
import './vds-poster.js';
import './vds-slider-value-text.js';
import './vds-slider-video.js';
import './vds-slider.js';
import './vds-time-slider.js';
import './vds-time.js';
import './vds-toggle-button.js';
import './vds-volume-slider.js';
