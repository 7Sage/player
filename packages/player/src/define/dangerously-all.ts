import './vds-audio-player.js';
import './vds-audio.js';
import './vds-hls-player.js';
import './vds-hls.js';
import './vds-media-controller.js';
import './vds-media-ui.js';
import './vds-media-sync.js';
import './vds-media-visibility.js';
import './vds-video-player.js';
import './vds-video.js';
import './dangerously-all-ui.js';
