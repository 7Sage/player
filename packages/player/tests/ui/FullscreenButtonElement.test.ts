import '$lib/define/vds-fullscreen-button';

import { elementUpdated } from '@open-wc/testing-helpers';
import { waitForEvent } from '@vidstack/foundation';
import { html } from 'lit';

import { buildMediaPlayerFixture } from '$test-utils';

async function buildFixture() {
  const { player } = await buildMediaPlayerFixture(html`
    <vds-fullscreen-button>
      <div class="enter"></div>
      <div class="exit"></div>
    </vds-fullscreen-button>
  `);

  const button = player.querySelector('vds-fullscreen-button')!;

  return { player, button };
}

it('should render light DOM', async function () {
  const { button } = await buildFixture();
  expect(button).dom.toMatchSnapshot();
});

it('should render shadow DOM', async function () {
  const { button } = await buildFixture();
  expect(button).shadowDom.to.equal(`<slot></slot>`);
});

it('should update `hidden` attribute based on fullscreen support', async () => {
  const { player, button } = await buildFixture();

  player._store.canFullscreen.set(true);
  await elementUpdated(button);
  expect(button.hasAttribute('hidden')).to.be.false;

  player._store.canFullscreen.set(false);
  await elementUpdated(button);
  expect(button.hasAttribute('hidden')).to.be.true;
});

it('should update fullscreen state', async () => {
  const { player, button } = await buildFixture();

  player._store.fullscreen.set(true);
  await elementUpdated(button);

  expect(button.pressed).to.be.true;
  expect(button.getAttribute('aria-pressed')).to.equal('true');
  expect(button.hasAttribute('media-fullscreen')).to.be.true;

  player._store.fullscreen.set(false);
  await elementUpdated(button);

  expect(button.pressed).to.be.false;
  expect(button.getAttribute('aria-pressed')).to.equal('false');
  expect(button.hasAttribute('media-fullscreen')).to.be.false;
});

it('should enter fullscreen', async function () {
  const { player, button } = await buildFixture();

  const requestFullscreenSpy = vi
    .spyOn(player, 'requestFullscreen')
    .mockImplementation(() => Promise.resolve());

  player._store.fullscreen.set(false);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-enter-fullscreen-request');
  expect(requestFullscreenSpy).toHaveBeenCalledOnce();
});

it('should exit fullscreen', async function () {
  const { player, button } = await buildFixture();

  const exitFullscreenSpy = vi
    .spyOn(player, 'exitFullscreen')
    .mockImplementation(() => Promise.resolve());

  player._store.fullscreen.set(true);
  await elementUpdated(button);

  setTimeout(() => button.dispatchEvent(new MouseEvent('pointerdown')));

  await waitForEvent(button, 'vds-exit-fullscreen-request');
  expect(exitFullscreenSpy).toHaveBeenCalledOnce();
});
