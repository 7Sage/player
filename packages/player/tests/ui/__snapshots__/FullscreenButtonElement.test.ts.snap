// Vitest Snapshot v1

exports[`should render light DOM 1`] = `
<vds-fullscreen-button
  aria-label="Fullscreen"
  aria-pressed="false"
  hidden=""
  role="button"
  tabindex="0"
>
  
      
  <div
    class="enter"
  />
  
      
  <div
    class="exit"
  />
  
    
</vds-fullscreen-button>
`;
